sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -0,0 +1,1422 @@
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual async Task<Response<DocumentResultCollection<Entity>>> RecognizePiiEntitiesAsync(IEnumerable<DocumentInput> inputs, TextAnalyticsRequestOptions options, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(inputs, nameof(inputs));
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope("Azure.AI.TextAnalytics.TextAnalyticsClient.RecognizePiiEntities");
+            scope.Start();
+
+            try
+            {
+                using Request request = CreateDocumentInputRequest(inputs, options, PiiEntitiesRoute);
+                Response response = await _pipeline.SendRequestAsync(request, cancellationToken);
+
+                return response.Status switch
+                {
+                    200 => await CreateRecognizeEntitiesResponseAsync(response, cancellationToken).ConfigureAwait(false),
+                    _ => throw await response.CreateRequestFailedExceptionAsync(),
+                };
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="inputs"></param>
+        /// <param name="options"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual Response<DocumentResultCollection<Entity>> RecognizePiiEntities(IEnumerable<DocumentInput> inputs, TextAnalyticsRequestOptions options, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(inputs, nameof(inputs));
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope("Azure.AI.TextAnalytics.TextAnalyticsClient.RecognizePiiEntities");
+            scope.Start();
+
+            try
+            {
+                using Request request = CreateDocumentInputRequest(inputs, options, PiiEntitiesRoute);
+                Response response = _pipeline.SendRequest(request, cancellationToken);
+
+                return response.Status switch
+                {
+                    200 => CreateRecognizeEntitiesResponse(response),
+                    _ => throw response.CreateRequestFailedException(),
+                };
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        #endregion
+
+        #region Entity Linking
+
+        /// <summary>
+        /// </summary>
+        /// <param name="inputText"></param>
+        /// <param name="language"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual async Task<Response<IEnumerable<LinkedEntity>>> ExtractEntityLinkingAsync(string inputText, string language = "en", CancellationToken cancellationToken = default)
```

Comment:
The type is `LinkedEntity` but the method is using "EntityLinking". This doesn't have a super strong correlation, and the method name sounds weird. Perhaps, for example, "ExtractLinkedEntities(Async)"?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8781